include/drivers/gpio.h

```diff
@@ -779,38 +779,28 @@ static inline int z_impl_gpio_port_toggle_bits(struct device *port,
 /**
  * @brief Set physical level of selected output pins.
  *
- * Pins which are simultaneously selected by set_pins and clear_pins value will
- * be toggled.
+ * Pins that are selected by both set_pins and clear_pins will be
+ * cleared to low.
```

Comment:
I believe the API would be "cleaner" if the function disallowed to have a pin selected by both set_pins and clear_pins values. If the application asks to have a pin set and cleared at the same time something clearly went wrong. It's better to report an error (or catch the issue via ASSERT) rather than quietly clear the pin.

Reply:
I think that'd be OK.  So the idea would be something like:

    __ASSERT((set_pins & clear_pins) == 0,
             "Pins cannot be both set and cleared");
    return gpio_port_set_masked(port, set_pins | clear_pins, set_pins);

The effect in the default situation of ignoring `__ASSERT` is that pins that are both set and cleared are actually just set, which is as good a choice as clearing them, and because this is done in the common header inline function such (illegal) usage would produce consistent behavior across drivers, which is desirable.

Reply:
OK, I'll implement the change, you don't need to push an update to your PR.